assets/prototype/application/services/apollo/relations/types.ts

```diff
@@ -0,0 +1,38 @@
+interface CommonProps {
+	relation?: string;
+	relationId?: string;
+	relationIds?: string[];
+}
+export interface RelationFunctionProps extends CommonProps {
+	entity: string;
+	entityId: string;
+}
+export interface RelationAction extends CommonProps {
+	type: string;
+	entity?: string;
+	entityId?: string;
+	data?: any;
+}
+type PossibleRelation = {
+	datetimes?: string[];
+	tickets?: string[];
+	prices?: string[];
+	priceTypes?: string[];
+};
```

Comment:
not sure what to think about this... but it seems like it would be tricky to solve in a more exact manner and I certainly can not think of a solution

Reply:
this sounds like we need to track this specific case via a tech debt issue, I've created that:
https://github.com/eventespresso/event-espresso-core/issues/2064
***
if you think that is obsolete or too much than that issue can be removed